src/views/List.jsx

```diff
@@ -1,16 +1,47 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 /** List component that displays items in a user's shopping cart  */
 export function List({ data }) {
+	const [filteredList, setFilteredList] = useState([]);
+	const [filterInput, setFilterInput] = useState('');
+
+	/* Use handler to change the state of filterInput  and convert all items to lowercase to facilitate a more thorough search */
+	const handleInput = (event) => {
+		const value = event.target.value.toLowerCase();
+		setFilterInput(value);
+		setFilteredList(
+			data.filter((item) => {
+				return item.name && item.name.toLowerCase().includes(value);
+			}),
+		);
+	};
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+
+			<form>
+				<label htmlFor="list-filter">Filter items</label>
+				<br />
+				<input
+					id="list-filter"
+					type="search"
+					placeholder="Start typing here..."
+					onChange={handleInput}
+				/>
+			</form>
+			{/* Uses data or state of filteredList depending on state of filterInput */}
 			<ul>
-				{data.map((item) => {
-					return <ListItem key={item.id} name={item.name} />;
-				})}
+				{!filterInput
+					? data.map((item) => {
+							return <ListItem key={item.id} name={item.name} />;
+					  })
+					: filteredList.map((item) => {
+							return <ListItem key={item.id} name={item.name} />;
```

Comment:
I know it's not part of the Acceptance Criteria, but it would be neat to check for an empty `filteredList` and display a message telling the user that no item exists in the shopping list.
 
![search-filter](https://user-images.githubusercontent.com/102492480/215049435-9d5b54b8-1e21-4b53-ab1f-dc622270a87e.gif)

Reply:
I think we're on the same page, so it would be like this:
Filter State | Count of Items | Message
---|---|---
Filtering|0|No items matching your search
Filtering|>0|(showing items)
Not Filtering|0|No items in your list
Not Filtering|>0|(showing items)

Reply:
Love this suggestion as well! And Matt, that breakdown is really helpful!

Reply:
@mtfoley i would love for you to do my code reviews 😀